diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7de7d37..a92f37e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -234,10 +234,19 @@ endif (IOS)
 unset(CERES_COMPILE_OPTIONS)
 message("-- Building with C++${CMAKE_CXX_STANDARD}")
 
+
 # Eigen.
 # Eigen delivers Eigen3Config.cmake since v3.3.3
-find_package(Eigen3 3.3 CONFIG REQUIRED
-  HINTS ${HOMEBREW_INCLUDE_DIR_HINTS})
+#####################
+# Patch comment
+# While it is true that Eigen3Config.cmake is
+# delivered since v3.3.3, it is not bundled with
+# the conan version. This patch enables compatibility
+# with conan.
+####################
+find_package(Eigen3 3.3 REQUIRED)
+set(EIGEN3_VERSION_STRING ${EIGEN3_VERSION})
+set(EIGEN3_INCLUDE_DIRS ${EIGEN3_INCLUDE_DIR})
 if (EIGEN3_FOUND)
   message("-- Found Eigen version ${EIGEN3_VERSION_STRING}: ${EIGEN3_INCLUDE_DIRS}")
   if (CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*)" AND
