diff --git a/CMakeLists.txt b/CMakeLists.txt
index 948bddb..e4217d8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -239,9 +239,14 @@ message("-- Building with C++${CMAKE_CXX_STANDARD}")
 
 # Eigen.
 # Eigen delivers Eigen3Config.cmake since v3.3.3
-find_package(Eigen3 3.3 CONFIG REQUIRED
-  HINTS ${HOMEBREW_INCLUDE_DIR_HINTS})
-if (EIGEN3_FOUND)
+find_package(Eigen3 3.3 REQUIRED)
+set(EIGEN3_VERSION_STRING ${EIGEN3_VERSION})
+set(EIGEN3_INCLUDE_DIRS ${EIGEN3_INCLUDE_DIR})
+if(NOT TARGET Eigen3::Eigen)
+   add_library(Eigen3::Eigen INTERFACE IMPORTED)
+   target_include_directories(Eigen3::Eigen SYSTEM INTERFACE ${EIGEN3_INCLUDE_DIR})
+endif()
+if (EIGEN3_FOUND OR Eigen3_FOUND)
   message("-- Found Eigen version ${EIGEN3_VERSION_STRING}: ${EIGEN3_INCLUDE_DIRS}")
   if (CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*)" AND
       EIGEN3_VERSION_STRING VERSION_LESS 3.3.4)
@@ -260,7 +265,7 @@ if (EIGEN3_FOUND)
     message("   which can still use the EIGEN_SPARSE_QR algorithm.")
     add_definitions(-DEIGEN_MPL2_ONLY)
   endif (EIGENSPARSE)
-endif (EIGEN3_FOUND)
+endif ()
 
 if (LAPACK)
   find_package(LAPACK QUIET)
