diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7de7d37..0789d9c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -236,6 +236,16 @@ message("-- Building with C++${CMAKE_CXX_STANDARD}")
 
 # Eigen.
 # Eigen delivers Eigen3Config.cmake since v3.3.3
+#####################
+# Patch comment
+# While it is true that Eigen3Config.cmake is
+# delivered since v3.3.3, it is not bundled with
+# the conan version. This patch enables compatibility
+# with conan.
+####################
+find_package(Eigen3 3.3 REQUIRED)
+set(EIGEN3_VERSION_STRING ${EIGEN3_VERSION})
+set(EIGEN3_INCLUDE_DIRS ${EIGEN3_INCLUDE_DIR})
 find_package(Eigen3 3.3 CONFIG REQUIRED
   HINTS ${HOMEBREW_INCLUDE_DIR_HINTS})
 if (EIGEN3_FOUND)
